nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java

```diff
@@ -0,0 +1,613 @@
+                    }
+
+                    boolean stop = false;
+
+                    while (!stop) {
+                        try {
+
+                            final List<Object> list = listReader.read(cellProcs);
+                            stop = list == null;
+
+                            if(!isWholeFFValidation && !stop) {
+                                validFF.set(session.append(validFF.get(), new OutputStreamCallback() {
+                                    @Override
+                                    public void process(OutputStream out) throws IOException {
+                                        out.write(print(list, csvPref, isFirstLine.get()));
+                                    }
+                                }));
+                                okCount.set(okCount.get() + 1);
+
+                                if(isFirstLine.get()) {
+                                    isFirstLine.set(false);
+                                }
+                            }
+
+                        } catch (final SuperCsvCellProcessorException e) {
+                            valid.set(false);
+                            if(isWholeFFValidation) {
+                                logger.debug("Failed to validate {} against schema due to {}; routing to 'invalid'", new Object[]{flowFile}, e);
+                                break;
+                            } else {
+                                // we append the invalid line to the flow file that will be routed to invalid relationship
+                                invalidFF.set(session.append(invalidFF.get(), new OutputStreamCallback() {
+                                    @Override
+                                    public void process(OutputStream out) throws IOException {
+                                        out.write(print(e.getCsvContext().getRowSource(), csvPref, isFirstLine.get()));
+                                    }
+                                }));
+
+                                if(isFirstLine.get()) {
+                                    isFirstLine.set(false);
+                                }
+                            }
+                        } finally {
+                            if(!isWholeFFValidation) {
+                                totalCount.set(totalCount.get() + 1);
+                            }
+                        }
+                    }
+
+                } catch (final IOException e) {
+                    valid.set(false);
+                    logger.error("Failed to validate {} against schema due to {}", new Object[]{flowFile}, e);
+                } finally {
+                    if(listReader != null) {
+                        listReader.close();
+                    }
+                }
+            }
+        });
+
+        if(isWholeFFValidation) {
+            if (valid.get()) {
+                logger.info("Successfully validated {} against schema; routing to 'valid'", new Object[]{flowFile});
```

Comment:
This line is probably too common to be "info". Debug would match the other paths in this block too.